api/ccache.go

```diff
@@ -0,0 +1,84 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"sync"
+
+	"github.com/raintank/metrictank/api/middleware"
+	"github.com/raintank/metrictank/api/models"
+	"github.com/raintank/metrictank/api/response"
+	"github.com/raintank/metrictank/cluster"
+	"github.com/raintank/worldping-api/pkg/log"
+)
+
+func (s *Server) ccacheDelete(ctx *middleware.Context, req models.CCacheDelete) {
+	res := models.CCacheDeleteResp{}
+
+	if req.Propagate {
+		res.Peers = s.ccacheDeletePropagate(ctx.Req.Context(), &req)
+	}
+
+	for _, pattern := range req.Patterns {
+		nodes, err := s.MetricIndex.Find(req.OrgId, pattern, 0)
+		if err != nil {
+			response.Write(ctx, response.NewError(http.StatusBadRequest, err.Error()))
+			return
+		}
+		for _, node := range nodes {
+			for _, def := range node.Defs {
+				delResult := s.Cache.DelMetric(def.Id)
+				res.DeletedSeries += delResult.Series
+				res.DeletedArchives += delResult.Archives
+			}
+		}
+	}
+	response.Write(ctx, response.NewJson(200, res, ""))
```

Comment:
if the peers reported a bunch of errors, or could not connect to peers, we should probably not report 200 ok